web/src/components/Calendar.js

```diff
@@ -0,0 +1,307 @@
+    <Appointments.Appointment {...restProps} className={classes.appointment} />
+));
+
+const AppointmentContent = withStyles(styles, {
+    name: 'AppointmentContent',
+})(({ classes, ...restProps }) => (
+    <Appointments.AppointmentContent
+        {...restProps}
+        className={classes.apptContent}
+    />
+));
+
+const FlexibleSpace = withStyles(styles, { name: 'ToolbarRoot' })(
+    ({ classes, ...restProps }) => (
+        <Toolbar.FlexibleSpace {...restProps} className={classes.flexibleSpace}>
+            <div className={classes.flexContainer}>
+                <Typography variant="h5" style={{ marginLeft: '10px' }}>
+                    My Calendar
+                </Typography>
+            </div>
+        </Toolbar.FlexibleSpace>
+    )
+);
+
+const getRandomColor = () => {
+    var letters = '0123456789ABCDEF';
+    var color = '#';
+    for (var i = 0; i < 6; i++) {
+        color += letters[Math.floor(Math.random() * 16)];
+    }
+    return color;
+};
+let setColor = false;
+
+const Calendar = ({ appointments }) => {
+    const [data, setData] = useState([]);
+    const [owners, setOwners] = useState([]);
+    const newDate = new Date();
+
+    const commitDeletes = ({ deleted }) => {
+        axios.delete(`/appointment/${deleted}`).then(response => {
+            if (response.status === 200) {
+                setData(data => data.filter(({ id }) => id !== deleted));
+                alert(`Appointment Successfully Deleted`);
+            }
+        });
+    };
+
+    const passInfo = () => {
+        setOwners([]);
+        setData([]);
+        appointments?.forEach((apt, i) => {
+            const newOwner = {
+                text: apt.provider,
+                id: i,
+                color: getRandomColor(),
+            };
+            setOwners(owners => [...owners, newOwner]);
+            axios.get(`/services/${apt.service_id}`).then(response => {
+                const service = response.data.result;
+                const startDate = new Date(apt.booked_time);
+                const endDate = new Date(startDate);
+                endDate.setTime(
+                    endDate.getTime() + service.duration * 60 * 1000
+                );
+                const tempObj = {
+                    id: apt._id,
+                    title: service.name,
+                    startDate: startDate,
+                    endDate: endDate,
+                    ownerId: i,
+                };
+                setData(data => [...data, tempObj]);
+            });
+        });
+        setColor = false;
+    };
+    if (!setColor && appointments.length === owners.length) {
+        owners?.forEach(name => {
+            for (let i = 1; i < owners.length; i++) {
+                if (
+                    name.text === owners[i].text &&
+                    owners[i].color !== name.color
+                ) {
+                    owners[i].color = name.color;
+                }
+            }
+        });
```

Comment:
this block of code doesn't really make sense to me. 

Why is this a nested loop?

Reply:
freakshow — Today at 8:52 PM
so this right here goes through each saved  appointment currently and looks at the provider associated with each. Since when we pull the the appointments we need to save each in its own index there will be some appointments with the same provider. so the if checks if the names match aka our saved name and name of appointment and the colors are not the same (since they are randomized) then it will change it so they match.